doc/design/status_port.md

```diff
@@ -0,0 +1,453 @@
+* Has to be attachable to `Listener`
+* `StatusPort` could potentially be used to communicate between different ASIL domains
+
+Potential applications are:
+
+* Introspection topics
+* Service discovery
+
+## Design
+
+### Terminology
+
+* Transaction: Atomic state of the world, which is changed by a write operation
+* Current transaction: Transaction in local scope, read in the beginning of each operation
+* Latest transaction: Transaction in the shared memory managment segment
+* Chunk: Untyped piece of memory located in the shared memory payload segment
+* Read position: The chunk, which was used the last time to write data
+* Write position: The opposite chunk, not currently being used by the `StatusPortReader`s
+
+### Contract & properties
+
+* 1:n communication aka broadcast
+* Two memory chunks from shared memory payload segment are used
+* Data exchanged needs to be trivially-copyable
+* Storing a chunk cannot fail
+* Reading a chunk cannot fail
+    * If no data was sent yet, `callable` is not called
+    * If data was updated in-between read, re-call `callable`
+* Not part of the user API in the first version, only used internally
+    * Might become user API for freedom-from-interference separation
+* Reader only needs write access to the data structure and no read access
+* Writing and Reading will be tried indefinitely till possible, hence starvation
+is possible
+* `StatusPortData` is created in the shared memory segment if either a `StatusPortWriter`
+or `StatusPortReader` is created
+    * Two chunks in the shared memory payload segment are bound to the lifetime
+    of the `StatusPortData` (either via `StatusPortData` c'tor or
+    `StatusPort{Writer,Reader}` c'tor)
+
+### Discarded ideas & design alternatives
+
+#### Atomic pointer
+
+The pointer to the currently active chunk could also be stored in an `std::atomic`.
+
+```cpp
+std::atomic<T*> activeChunk{nullptr}
+```
+
+However, it would need the full 64-bit and which is not needed when managing
+just two chunks. Hence an `abaCounter` would need to be stored in a separate
+`std::atomic` variable.
+
+#### Omit `ServiceDescription` and discovery
+
+Methods like `offer()` or `stopOffer()` are not needed to keep things simple.
+However, omitting the `ServiceDescription` completely is not possible as it is
+needed to associate the right `StatusPortData` in shared memory with the correct
+writer or reader object on the stack. The `StatusPort` shall follow the
+service-based design.
+
+#### `atomic<T>::exchange` in `StatusPortReader::take()`
+
+A [CAS](https://en.wikipedia.org/wiki/Compare-and-swap) operation is not possible
+because CAS needs write access and a `StatusPortReader` shall not be allowed to
+write data into shared memory.
+
+#### Differentiate between Writer/Reader between RouDi/User?
+
+To be consistent with the previously available publisher and subscribers one
+could argue to follow the same naming. However, the `StatusPort` might be used to
```

Comment:
I doubt that the `StatusPort` can be used to communicate between ASIL domains since it has some very heavy restrictions on the data.
* self contained
* trivially copyable
* changes never or only rarely
* no history possible
* can be corrupted when reading (when something writes new data)

Reply:
You're definitely right when it comes to the restrictions. However, my current view is that it might be the better approach than other alternatives to separate ASIL partitions. Needs to be seen.